ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ConnectorDescriptionImpl.java

```diff
@@ -40,18 +42,20 @@ public ConnectorDescriptionImpl(
   ) {
     this.name = Objects.requireNonNull(name, "name");
     this.className = Objects.requireNonNull(className, "className");
-    this.sources = Objects.requireNonNull(sources, "sources");
-    this.topics = Objects.requireNonNull(topics, "topics");
+    this.sources = ImmutableList.copyOf(Objects.requireNonNull(sources, "sources"));
+    this.topics = ImmutableList.copyOf(Objects.requireNonNull(topics, "topics"));
     this.type = Objects.requireNonNull(type, "type");
     this.state = Objects.requireNonNull(state, "state");
   }
 
   @Override
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "sources is ImmutableList")
   public List<String> sources() {
     return sources;
   }
 
   @Override
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "topics is ImmutableList")
```

Comment:
Can we return a deep copy instead of suppressing? 

`List.copyOf(topics)`

Reply:
We do return an `ImmutableList` do what do we gain by making a deep-copy? The issue with `ImmutableList` is, that spotbug does not detect it as immutable, and thus we need to add the suppress.

I agree that it's annoying that we need to annotation, and we could also return a deep-copy, but it seems unnecessary?

Reply:
Sounds good. If https://github.com/spotbugs/spotbugs/issues/1601 ever gets resolved, we should remove all the annotations so that we can differentiate the actual mutable methods from ones the ones that use Guava classes.